pydantic_ai_slim/pydantic_ai/_parts_manager.py

```diff
@@ -75,82 +224,254 @@ def handle_text_delta(
+
+            if after_start:
+                yield from self.handle_thinking_delta(vendor_part_id=vendor_part_id, content=after_start)
+            return
+
+        # emit as TextPart
+        yield from self._emit_text_part(
+            vendor_part_id=vendor_part_id,
+            content=content,
+            id=id,
+            ignore_leading_whitespace=ignore_leading_whitespace,
+        )
+
+    def _handle_text_delta_with_thinking_tags(
+        self,
+        *,
+        vendor_part_id: VendorId,
+        content: str,
+        id: str | None,
+        thinking_tags: tuple[str, str],
+        ignore_leading_whitespace: bool,
+    ) -> Generator[ModelResponseStreamEvent, None, None]:
+        """Handle text delta with thinking tag detection and buffering for split tags."""
+        start_tag, end_tag = thinking_tags
+        buffered = self._thinking_tag_buffer.get(vendor_part_id, '')
+
+        part_index = self._vendor_id_to_part_index.get(vendor_part_id)
+        existing_part = self._parts[part_index] if part_index is not None else None
+
+        # If a TextPart has already been created for this vendor_part_id, disable thinking tag detection
+        if existing_part is not None and isinstance(existing_part, TextPart):
+            combined_content = buffered + content
+            self._thinking_tag_buffer.pop(vendor_part_id, None)
+            yield from self._emit_text_part(
+                vendor_part_id=vendor_part_id,
+                content=combined_content,
+                id=id,
+                ignore_leading_whitespace=False,
+            )
+            return
+
+        in_thinking = existing_part is not None and isinstance(existing_part, ThinkingPart)
+
+        segments, new_buffer = _parse_chunk_for_thinking_tags(
+            content=content,
+            buffered=buffered,
+            start_tag=start_tag,
+            end_tag=end_tag,
+            in_thinking=in_thinking,
+        )
+
+        # Check for text before thinking tag - if so, treat entire combined content as text
+        if segments and segments[0][0] == 'text':
+            text_content = segments[0][1]
+            if ignore_leading_whitespace and text_content.isspace():
+                text_content = ''
+
+            if text_content:
+                combined_content = buffered + content
+                self._thinking_tag_buffer.pop(vendor_part_id, None)
+                yield from self._emit_text_part(
+                    vendor_part_id=vendor_part_id,
+                    content=combined_content,
+                    id=id,
+                    ignore_leading_whitespace=False,
+                )
+                return
+
+        for i, (segment_type, segment_content) in enumerate(segments):
+            if segment_type == 'text':
+                # Skip whitespace-only text before a thinking tag when ignore_leading_whitespace=True
+                skip_whitespace_before_tag = (
+                    ignore_leading_whitespace
```

Comment:
Maybe it would make sense to move this whitespace logic to the chunking function